pkg/xds/auth/k8s/authenticator.go

```diff
@@ -103,7 +91,7 @@ func (k *kubeAuthenticator) authResource(ctx context.Context, resource model.Res
 		return err
 	}
 
-	if err := k.verifyToken(ctx, credential, namespace, serviceAccountName); err != nil {
+	if err := k.verifyToken(ctx, credential, namespace, serviceAccountName, resource.GetMeta().GetName()); err != nil {
```

Comment:
This verifyToken is a level of indirection that I find only makes things hard to read don't you think? Should we inline this method instead?

Reply:
Yeah, we can do it. To me, it looks more like this `authResource` method is inlined.

Reply:
Yeah one way or the other :) Just feels that the nesting is just making things harder to read.